projects/core/components/svg/svg.component.ts

```diff
@@ -40,10 +43,15 @@ const FAILED_EXTERNAL_ICON = 'Failed to load external SVG';
     templateUrl: './svg.template.html',
     styleUrls: ['./svg.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [],
 })
 export class TuiSvgComponent {
+    private readonly srcProcessors = tuiFlatten(
```

Comment:
Why do we need to flatten them?

Reply:
`Why do we need to flatten them?` without it I get array in array error

Reply:
I don't get it. How does it become multi-dimensional array?

Reply:
Got it. This way if you provide multiple processors, each of them would have all the parent processors attached. Do we really need to inherit from parent here?

Reply:
`Do we really need to inherit from parent here?`  yes

otherwise component provider will override the default `tuiSvgMultiSrcProcessor` processors that can be provided by us in `proprietary-root.module.ts` or `root.module.ts`

Reply:
We can document it as a global token maybe? Is there any merit to hierarchy in its case?

Reply:
`TUI_SVG_SRC_PROCESSOR` was a global token 

and thus it causes some problems where users can override the default behaviour which may be necessary

Reply:
fixed